pages/post/[postId].js

```diff
@@ -22,18 +22,16 @@ export async function getServerSideProps(context) {
     let post = await GetstreamSingleton.getInstance().getPostById(originalPostId)
 
     if (!PostUtil.isPostPublic(post)) {
-        console.log('masuk private')
         let redirect = await RedirectUtils.redirectPrivatePost(userAgent, post)
         if (redirect) return redirect
     }
 
     if (DateUtils.isPostExpired(post)) {
-        console.log('masuk expired')
         let redirect = await RedirectUtils.redirectExpiredPost(userAgent)
         if (redirect) return redirect
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code checks if the post is private or expired and then redirects accordingly. However, there's no error handling in case `RedirectUtils.redirectPrivatePost(userAgent, post)` or `RedirectUtils.redirectExpiredPost(userAgent)` fails. Consider adding a try-catch block to handle potential errors.
```diff
     if (!PostUtil.isPostPublic(post)) {
+        try {
             let redirect = await RedirectUtils.redirectPrivatePost(userAgent, post)
             if (redirect) return redirect
+        } catch (error) {
+            console.error('Error while redirecting private post:', error);
+        }
     }
  
     if (DateUtils.isPostExpired(post)) {
+        try {
             let redirect = await RedirectUtils.redirectExpiredPost(userAgent)
             if (redirect) return redirect
+        } catch (error) {
+            console.error('Error while redirecting expired post:', error);
+        }
     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->